src/test/kotlin/org/rust/lang/core/type/RsGenericExpressionTypeInferenceTest.kt

```diff
@@ -241,6 +241,96 @@ class RsGenericExpressionTypeInferenceTest : RsTypificationTestBase() {
         }
     """)
 
+    fun `test self 1`() = testExpr("""
+        struct Foo(u8);
+        fn main() { }
+        impl Foo {
+            fn foo(self) {
+                self;
+              //^ Foo
+            }
+        }
+    """)
+
+    fun `test self 2`() = testExpr("""
+        struct Foo(u8);
+        fn main() { }
```

Comment:
Why we need `fn main() { }` and that `u8` field?

Reply:
Not needed, just copy paste leftovers.
Should be fixed now.